File: hospital-backend/KillLocks.java
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.ResultSet;
import java.sql.Statement;

public class KillLocks {
    public static void main(String[] args) throws Exception {
        String url = "jdbc:postgresql://db.lvaymongduklhaaklomk.supabase.co:5432/postgres";
        String user = "postgres";
        String password = "SumitDubey@1234";

        try (Connection conn = DriverManager.getConnection(url, user, password);
             Statement stmt = conn.createStatement()) {
            
            System.out.println("Finding locked connections...");
            ResultSet rs = stmt.executeQuery("SELECT pid, state, query FROM pg_stat_activity WHERE state LIKE 'idle in transaction%'");
            
            while (rs.next()) {
                int pid = rs.getInt("pid");
                String state = rs.getString("state");
                String query = rs.getString("query");
                System.out.println("Killing PID " + pid + " (State: " + state + ", Query: " + query + ")");
                
                try (Statement killStmt = conn.createStatement()) {
                    killStmt.execute("SELECT pg_terminate_backend(" + pid + ")");
                }
            }
            System.out.println("Done.");
        }
    }
}

File: hospital-backend/kill_locks.js
const { Client } = require('pg');

async function run() {
    const client = new Client({
        connectionString: "postgresql://postgres:SumitDubey@1234@db.lvaymongduklhaaklomk.supabase.co:5432/postgres"
    });
    
    await client.connect();
    
    console.log("Finding locked connections...");
    const res = await client.query("SELECT pid, state, query FROM pg_stat_activity WHERE state LIKE 'idle in transaction%'");
    
    for (const row of res.rows) {
        console.log(`Killing PID ${row.pid} (State: ${row.state}, Query: ${row.query})`);
        await client.query(`SELECT pg_terminate_backend(${row.pid})`);
    }
    
    console.log("Done. Locks cleared.");
    await client.end();
}

run().catch(console.error);
